frontend/widgets/VolumeMeter.hpp

```diff
@@ -37,38 +37,23 @@ class VolumeMeter : public QWidget {
 	Q_PROPERTY(QColor foregroundErrorColorDisabled READ getForegroundErrorColorDisabled WRITE
 			   setForegroundErrorColorDisabled DESIGNABLE true)
 
-	Q_PROPERTY(QColor clipColor READ getClipColor WRITE setClipColor DESIGNABLE true)
 	Q_PROPERTY(QColor magnitudeColor READ getMagnitudeColor WRITE setMagnitudeColor DESIGNABLE true)
 	Q_PROPERTY(QColor majorTickColor READ getMajorTickColor WRITE setMajorTickColor DESIGNABLE true)
 	Q_PROPERTY(QColor minorTickColor READ getMinorTickColor WRITE setMinorTickColor DESIGNABLE true)
-	Q_PROPERTY(int meterThickness READ getMeterThickness WRITE setMeterThickness DESIGNABLE true)
-	Q_PROPERTY(qreal meterFontScaling READ getMeterFontScaling WRITE setMeterFontScaling DESIGNABLE true)
-
-	// Levels are denoted in dBFS.
-	Q_PROPERTY(qreal minimumLevel READ getMinimumLevel WRITE setMinimumLevel DESIGNABLE true)
-	Q_PROPERTY(qreal warningLevel READ getWarningLevel WRITE setWarningLevel DESIGNABLE true)
-	Q_PROPERTY(qreal errorLevel READ getErrorLevel WRITE setErrorLevel DESIGNABLE true)
-	Q_PROPERTY(qreal clipLevel READ getClipLevel WRITE setClipLevel DESIGNABLE true)
-	Q_PROPERTY(qreal minimumInputLevel READ getMinimumInputLevel WRITE setMinimumInputLevel DESIGNABLE true)
 
-	// Rates are denoted in dB/second.
-	Q_PROPERTY(qreal peakDecayRate READ getPeakDecayRate WRITE setPeakDecayRate DESIGNABLE true)
+	friend class VolControl;
 
-	// Time in seconds for the VU meter to integrate over.
-	Q_PROPERTY(qreal magnitudeIntegrationTime READ getMagnitudeIntegrationTime WRITE setMagnitudeIntegrationTime
-			   DESIGNABLE true)
+private:
+	OBSWeakSource weakSource;
+	OBSVolMeter obs_volmeter;
```

Comment:
```suggestion
	OBSVolMeter obsVolumeMeter;
```

I take it `OBSVolMeter` is a RAII wrapper and thus cannot easily be renamed to `OBSVolumeMeter`?

I also think we're running into confusions with proper encapsulation classes like `OBS::VolumeMeter` and the associated RAII wrappers like `OBSVolMeter` here. The same problem applies to canvases, because if done right the RAII wrappers should not exist:

The `VolumeMeter` should be the one and only representation of a volume meter and handles resource management, lifetime, and thus needs to be properly owned and ownership of it transferred.